internal/planner/filter/traverse.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+package filter
+
+import (
+	"github.com/sourcenetwork/defradb/client/request"
+	"github.com/sourcenetwork/defradb/internal/connor"
+	"github.com/sourcenetwork/defradb/internal/planner/mapper"
+)
+
+// TraverseFields walks through a filter condition tree and calls the provided function f
+// for each leaf node (field value) encountered. The function f receives the path to the field
+// (as a string slice) and its value. If f returns false, traversal stops immediately.
+//
+// The path parameter in f represents the nested field names leading to the value, excluding
+// operator keys (those starting with '_'). For example, given the filter:
+//
+//	{
+//	    "author": {
+//	        "books": {
+//	            "title": {"_eq": "Sample"}
+//	        }
+//	    }
+//	}
+//
+// The callback would receive path=["author", "books", "title"] and value="Sample"
+func TraverseFields(conditions map[string]any, f func([]string, any) bool) {
```

Comment:
todo: The return value of this function is unused, and unexplained by the documentation.  Please remove it.